cpp/tests/test_save_results.cpp

```diff
@@ -339,4 +370,100 @@ TEST(TestSaveResult, save_percentiles_and_sums)
+
+    auto result_edges_from_file = results_edges_from_file_p25.value()[0];
+    EXPECT_EQ(ensemble_edges.back().back().get_num_elements(), result_edges_from_file.get_groups().get_num_elements());
+    EXPECT_EQ(ensemble_edges.back().back().get_num_time_points(),
+              result_edges_from_file.get_groups().get_num_time_points());
+
+    // single runs
+    auto results_edges_run0 = mio::read_result(tmp_results_dir + "/Edges_run0.h5");
+    ASSERT_TRUE(results_edges_run0);
+    auto results_edges_run1 = mio::read_result(tmp_results_dir + "/Edges_run1.h5");
+    ASSERT_TRUE(results_edges_run1);
+    auto results_edges_run2 = mio::read_result(tmp_results_dir + "/Edges_run2.h5");
+    ASSERT_TRUE(results_edges_run2);
 }
+
+TEST(TestSaveResult, save_edges)
+{
+    // create some results and pairs_edges
+    const auto n = Eigen::Index(3);
+    std::vector<mio::TimeSeries<double>> results_edges(3, mio::TimeSeries<double>(n));
+    results_edges[0].add_time_point(0.0, Eigen::VectorXd::Constant(n, 0));
+    results_edges[0].add_time_point(1.0, Eigen::VectorXd::Constant(n, 1));
+    results_edges[0].add_time_point(2.0, Eigen::VectorXd::Constant(n, 1));
+
+    results_edges[1].add_time_point(0.0, Eigen::VectorXd::Constant(n, 2));
+    results_edges[1].add_time_point(1.0, Eigen::VectorXd::Constant(n, 3));
+    results_edges[1].add_time_point(2.0, Eigen::VectorXd::Constant(n, 5));
+
+    results_edges[2].add_time_point(0.0, Eigen::VectorXd::Constant(n, 3));
+    results_edges[2].add_time_point(1.0, Eigen::VectorXd::Constant(n, 4));
+    results_edges[2].add_time_point(2.0, Eigen::VectorXd::Constant(n, 7));
+
+    const std::vector<std::pair<int, int>> pairs_edges = {{0, 1}, {0, 2}, {1, 2}};
+
+    // save the results to a file
+    TempFileRegister file_register;
+    auto results_file_path = file_register.get_unique_path("test_result-%%%%-%%%%.h5");
+    auto save_edges_status = mio::save_edges(results_edges, pairs_edges, results_file_path);
+    ASSERT_TRUE(save_edges_status);
+
+    // read the results back in and check that they are correct.
+    auto results_from_file = mio::read_result(results_file_path);
+    ASSERT_TRUE(results_from_file);
+
+    // group 0
+    auto result_from_file_group0 = results_from_file.value()[0];
+    ASSERT_EQ(result_from_file_group0.get_groups().get_num_time_points(), 3);
+    ASSERT_EQ(result_from_file_group0.get_groups().get_num_elements(), 6);
+    ASSERT_EQ(result_from_file_group0.get_groups().get_value(0), (Eigen::VectorXd(6) << 0, 0, 0, 2, 2, 2).finished());
+    ASSERT_EQ(result_from_file_group0.get_groups().get_value(1), (Eigen::VectorXd(6) << 1, 1, 1, 3, 3, 3).finished());
+    ASSERT_EQ(result_from_file_group0.get_groups().get_value(2), (Eigen::VectorXd(6) << 1, 1, 1, 5, 5, 5).finished());
+    ASSERT_EQ(result_from_file_group0.get_groups().get_time(0), 0.0);
+    ASSERT_EQ(result_from_file_group0.get_groups().get_time(1), 1.0);
+    ASSERT_EQ(result_from_file_group0.get_groups().get_time(2), 2.0);
```

Comment:
Please only use ASSERT_X when the test needs to stop on failure. Multiple EXPECT_X means potentially a lot more output on failure, but you usually want that to find the error/bug that caused the test to fail.